Parkinson/TasksAndSteps/APHParkinsonActivityViewController.m

```diff
@@ -0,0 +1,254 @@
+                                 NSLocalizedString(kMomentInDayFormatChoiceTookMyMedicine, nil),
+                                 NSLocalizedString(kMomentInDayFormatChoiceEvening, nil),
+                                 NSLocalizedString(kMomentInDayFormatChoiceNone, nil)
+                                 ];
+            ORKAnswerFormat  *format = [ORKTextChoiceAnswerFormat
+                                        choiceAnswerFormatWithStyle:ORKChoiceAnswerStyleSingleChoice
+                                                         textChoices:choices];
+            
+            ORKFormItem  *item = [[ORKFormItem alloc] initWithIdentifier:kMomentInDayFormat
+                                                                   text:NSLocalizedString(kMomentInDayFormatItemText, nil)
+                                                           answerFormat:format];
+            [stepQuestions addObject:item];
+        }
+        [step setFormItems:stepQuestions];
+        
+        NSMutableArray  *newSteps = [task.steps mutableCopy];
+        if ([newSteps count] >= 1) {
+            [newSteps insertObject:step atIndex:1];
+        }
+        replacementTask = [[ORKOrderedTask alloc] initWithIdentifier:taskTitle steps:newSteps];
+    }
+    return  replacementTask;
+}
+
+#pragma  mark  -  Step View Controller Delegate Methods
+
+- (void)stepViewControllerResultDidChange:(ORKStepViewController *)stepViewController
+{
+    NSString  *stepIdentifier = stepViewController.step.identifier;
+    if ([stepIdentifier isEqualToString:kMomentInDayStepIdentifier] == YES) {
+        ORKStepResult  *stepResult = stepViewController.result;
+        if ([stepResult.results count] > 0) {
+            id  object = [stepResult.results lastObject];
+            if ([object isKindOfClass:[ORKChoiceQuestionResult class]] == YES) {
+                ORKChoiceQuestionResult  *result = (ORKChoiceQuestionResult *)object;
+                    //
+                    //  stash the object away
+                    //
+                if (result.choiceAnswers.lastObject != nil) {
+                    NSDictionary  *dictionary = @{
+                                                  kMomentInDayChoiceAnswerKey : result.choiceAnswers.lastObject,
+                                                  kMomentInDayQuestionTypeKey : @(result.questionType),
+                                                  kMomentInDayIdentifierKey   : result.identifier,
+                                                  kMomentInDayStartDateKey    : result.startDate,
+                                                  kMomentInDayEndDateKey      : result.endDate,
+                                                  };
+                    NSUserDefaults  *defaults = [NSUserDefaults standardUserDefaults];
+                    [defaults setObject:dictionary forKey:kMomentInDayUserDefaultsKey];
+                    [defaults synchronize];
+                }
```

Comment:
> On Jun 29, 2015, at 2:43 PM, Yuan Zhu <notifications@github.com> wrote:
> 
> In Parkinson/TasksAndSteps/APHParkinsonActivityViewController.m <https://github.com/ResearchKit/mPower/pull/6#discussion_r33517299>:
> 
> > +                ORKChoiceQuestionResult  *result = (ORKChoiceQuestionResult *)object;
> > +                    //
> > +                    //  stash the object away
> > +                    //
> > +                if (result.choiceAnswers.lastObject != nil) {
> > +                    NSDictionary  *dictionary = @{
> > +                                                  kMomentInDayChoiceAnswerKey : result.choiceAnswers.lastObject,
> > +                                                  kMomentInDayQuestionTypeKey : @(result.questionType),
> > +                                                  kMomentInDayIdentifierKey   : result.identifier,
> > +                                                  kMomentInDayStartDateKey    : result.startDate,
> > +                                                  kMomentInDayEndDateKey      : result.endDate,
> > +                                                  };
> > +                    NSUserDefaults  *defaults = [NSUserDefaults standardUserDefaults];
> > +                    [defaults setObject:dictionary forKey:kMomentInDayUserDefaultsKey];
> > +                    [defaults synchronize];
> > +                }
> Thanks for explaining.
> 
> Can you use NSSecureCoding to save the stepResult into NSUserDefaults? 
> So you don't need to rebuild the object on restoring.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ResearchKit/mPower/pull/6/files#r33517299>.
> 
In principle, yes, we could replace that result —> dictionary and dictionary —> result.
Presumably you are suggesting creating an archive and stashing that in NSUserDefaults.

However, the code has been in shipping versions of the App for a couple of months now,
and so we would be faced with a migration path if we wanted to switch to the NSSecureCoding
version. 

    Best Wishes,
        . . . . . . . .    Henry